src/app/permissions/authorized-users-select/authorized-users-select.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { LayoutModule } from '@angular/cdk/layout';
+import { FlexLayoutModule } from '@angular/flex-layout';
+import { ReactiveFormsModule } from '@angular/forms';
+import { AppMaterialModule } from '../../app-material.module';
+import { User } from '../../users/user';
+import { MatAutocompleteSelectedEvent } from '@angular/material';
+
+const userSuggestions: User[] = [
+  {
+    id: '2736a204-f3ab-4b65-8568-a1c8ce1db8ab',
+    userName: 'testing',
+    firstName: 'Tina',
+    lastName: 'Testing',
+    email: 'tina.testing@skoop.io',
+    coach: false,
+  }
+];
+
+describe('AuthorizedUsersSelectComponent', () => {
+  let component: AuthorizedUsersSelectComponent;
+  let fixture: ComponentFixture<AuthorizedUsersSelectComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        BrowserAnimationsModule,
+        LayoutModule,
+        FlexLayoutModule,
+        ReactiveFormsModule,
+        AppMaterialModule
+      ],
+      declarations: [ AuthorizedUsersSelectComponent ],
+      providers: [
+        {
+          provide: UsersService,
+          useValue: jasmine.createSpyObj('userService', {
+            'getAuthorizedUsers': of<User[]>(
+              [{
+                id: '2736a204-f3ab-4b65-8568-a1c8ce1db8ab',
+                userName: 'testing',
+                firstName: 'Tina',
+                lastName: 'Testing',
+                email: 'tina.testing@skoop.io',
+                coach: false,
+              }]
+            ),
+          'getUserSuggestions': of<User[]>(userSuggestions)})
+        }
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AuthorizedUsersSelectComponent);
+    component = fixture.componentInstance;
+    component.users = [];
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should remove a given user from the list of authorized users', () => {
+    component.users = [{
+      id: '2736a204-f3ab-4b65-8568-a1c8ce1db8ab',
+      userName: 'testing',
+      firstName: 'Tina',
+      lastName: 'Testing',
+      email: 'tina.testing@skoop.io',
+      coach: false,
+    },
+      {
+        id: '251c2a3b-b737-4622-8060-196d5e297ebc',
+        userName: 'testbed',
+        firstName: 'Tabia',
+        lastName: 'Testbed',
+        email: 'tabia.testbed@skoop.io',
+        coach: false,
+      }];
+    component.onAuthorizedUserRemoved(component.users[0]);
+    expect(component.users).toEqual([{
+      id: '251c2a3b-b737-4622-8060-196d5e297ebc',
+      userName: 'testbed',
+      firstName: 'Tabia',
+      lastName: 'Testbed',
+      email: 'tabia.testbed@skoop.io',
+      coach: false,
+    }]);
+  });
+
+
+  it('should send getUserSuggestions request in 500 ms', fakeAsync(() => {
+    const usersService = TestBed.get(UsersService) as UsersService;
+
+    component.authorizedUsersControl.setValue('test');
+    tick(200);
```

Comment:
I would prefer to avoid usage of any precise time intervals like `200` or `300` and to simulate an arbitrary time interval (no matter how much time has passed).